app/lib/humane_transport_app.dart

```diff
@@ -17,6 +19,11 @@ class _HumaneTransportAppState extends State<HumaneTransportApp> {
       initialRoute: SplashScreen.route,
       onGenerateRoute: AppRouteGenerator.onGenerateRoute,
       navigatorKey: locator<NavigationService>().navigationKey,
+      builder: (context, child) => Navigator(
```

Comment:
The navigator is a dialog service???

Reply:
FilledStack - _It's common that the BuildContext is not available where you business logic is, whether you're using BLoC, pure provider, redux, get_it only, get, etc. It's easier in terms of writing code to be able to show your dialogs from where your failures / success are happening, which is why the DialogService was born._

Reply:
Can you send me the link you got this from? I can guess that the material app builder of widgets uses dialog manager somehow, but I don't understand how.

Reply:
I provided the link in my commit notes, but here it  is: [FilledStack](https://www.filledstacks.com/post/show-dialogs-without-context/)

Reply:
In the example in your link:
```
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: DialogExampleView(),
      onGenerateRoute: Router().onGenerateRoute,
      // Set the navigatorKey
      navigatorKey: locator<DialogService>().navigatorKey,
    );
  }
}
```

It looks like the navigator key is the dialog service's. What I still don't understand is that you've wrapped the dialog manager in a navigator. Why?

Reply:
Check this [article](https://medium.com/flutter-community/manager-your-flutter-dialogs-with-a-dialog-manager-1e862529523a) and it explains everything about dialog service and why he chose to wrap it with Navigator.

Reply:
Ah, OK. Makes sense now. Could you comment that article in this file?

Reply:
will do!